src/CommunicationLayer/CommunicationContainer.cpp

```diff
@@ -20,10 +20,10 @@ class CommunicationContainerPrivate
                                      infrastructureContainer.settings().queue(),
                                      infrastructureContainer.settings().ipAddress(),
                                      infrastructureContainer.settings().port())
-        , commDeviceManager_(udpSocket_)
+        , commDeviceManager_(internetConnectionService_.getChannel(),
```

Comment:
It makes more sense to have the `commDeviceManager` connect to the `connectionController`, and have `connectionController` call the `getChannel()` function. 

Since we're verifying the `type` in `commDeviceManager` to create the `InternetCommDevice`, I think we should do the same thing for the `connectionController` & the `InternetConnectionDevice`. The blue marker in this diagram is what I mean. Let me know if you disagree!
![image](https://user-images.githubusercontent.com/15304203/35466371-7e172304-02c0-11e8-83e2-354a625867ac.png)